### PR TITLE
[CUR-560] docx thread details

### PR DESCRIPTION
**Pull request by orangemug**
## Description

- Added thread details pages to curriculum generated `.docx` 

## Issue(s)

`CUR-560`

## How to test

1. Go to https://deploy-preview-2445--oak-web-application.netlify.thenational.academy/teachers/curriculum/docx-poc/select
2. Choose a subject via the autoComplete
3. Download the docx template - linked on the page
4. Download the rendered document
5. Check the cover page is present and correct

## Screenshots

Screenshot 2024-05-22 at 14 00 06

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/pages-helpers/curriculum/docx/curriculum-download-patcher/patches/mainThreads.ts [hunk 1]
```diff
@@ -1,119 +1,129 @@
...
+import { Element } from "xml-js";
 
 import { xmlElementToJson } from "../../xml";
 import { CombinedCurriculumData } from "..";
 
 import { createThreadOptions, threadUnitByYear } from "./util";
 
+import { Unit } from "@/components/CurriculumComponents/CurriculumVisualiser";
+
+function sortByOrder(units: Unit[]) {
```
Comment (sonalisin) [inline src/pages-helpers/curriculum/docx/curriculum-download-patcher/patches/mainThreads.ts:10]: Can we move this to the [tab].tsx and pull it through? that way it's not being repeated in [tab]. It's a note for me for testing as well as I'm using it in the test file

File: src/pages-helpers/curriculum/docx/curriculum-download-patcher/patches/util.ts [hunk 2]
```diff
@@ -30,20 +30,17 @@ export const textReplacer = (
...
+  const output = {} as Record<string, Unit[]>;
 
   units.forEach((unit: Unit) => {
     unit.threads.forEach((thread) => {
       if (thread.slug === threadSlug) {
         output[unit.year] = output[unit.year] ?? [];
         if (
           output[unit.year] &&
-          !output[unit.year]!.find(({ unit: u }) => u.slug === unit.slug)
+          !output[unit.year]!.find((u) => u.slug === unit.slug)
```
Comment (sonalisin) [inline src/pages-helpers/curriculum/docx/curriculum-download-patcher/patches/util.ts:41]: Can `u` be named something a bit clearer? Not sure what it's trying to do

**Review by sonalisin — APPROVED** [review]
Looks good other than 2 minor comments!

File: src/pages-helpers/curriculum/docx/curriculum-download-patcher/patches/mainThreads.ts (hunk 1 above)
Comment (orangemug) [reply src/pages-helpers/curriculum/docx/curriculum-download-patcher/patches/mainThreads.ts:10]: Going to leave this, I've added a ticket to next sprint to find a nice place for helper functions.

**Pull request merged**